src/functions-and-arrays.js

```diff
@@ -1,41 +1,141 @@
 
-function findLongestWord() {}
+function findLongestWord(words) {
+  let longestWord;
+  if (words.length === 0){
+    return null;
+
+  } else {
+    for (let i=0; i < words.length; i++){
+      let word = words[i];
+      if (!longestWord || longestWord.length < word.length){
+      longestWord = word;
+  }
+    } 
+    return longestWord;
+  }
+}
 
 
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
+function sumNumbers(numbers) {
+  let sum = 0
+  if (numbers.length === 0){
+    return 0;
+  } else {
+    for (let i = 0; i<numbers.length; i++){
+      let number = numbers[i]
+      sum += number;
+      }
+  }
+  return sum
 
+  }
+  
 
 
 // Iteration #3.1 Bonus:
-function sum() {}
+function sum(mixedArr) {
+  let countChar = 0;
+  let countNum = 0;
+  let countBol = 0;
+  
+  if (mixedArr.length === 0) {
+    return 0;
+  } else {
+    for (let i = 0; i < mixedArr.length; i++) {
+      let element = mixedArr[i];
+      if (typeof element === 'number') {
+        countNum += element;
+      } else if (typeof element === 'string') {
+        countChar += element.length;
+      } else if (typeof element === 'boolean'){
+        countBol += element;
+      } else {
+        throw new Error("Unsupported data type sir or ma'am");
+      }
+    }
+  }
+  let sumNumChar = countChar + countNum+ countBol;
+  return sumNumChar;
+}
 
 
 
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
-function averageNumbers() {}
+function averageNumbers(numbersAvg) {
+  let sum = 0
+  if (numbersAvg.length === 0){
+    return null;
+  } else {
+    for (let i = 0; i < numbersAvg.length; i++) {
+      let number = numbersAvg[i];
+      sum += number; 
+      }
+  }
+
+  return sum / numbersAvg.length;
+}
 
 
 // Level 2: Array of strings
 const wordsArr = ['seat', 'correspond', 'linen', 'motif', 'hole', 'smell', 'smart', 'chaos', 'fuel', 'palace'];
 
-function averageWordLength() { }
+function averageWordLength(wordsArr) { 
+  let count = 0;
+ if (wordsArr.length === 0){
+  return null;
+ } else {
+  for (let i = 0; i < wordsArr.length; i++) {
+    let char = wordsArr[i];
+    count += char.length;
+  }
+ }
+  return count / wordsArr.length;
+}
+
 
 // Bonus - Iteration #4.1
-function avg() {}
+function avg(wordsArr) {
+  let countChar = 0;
+  let countNum = 0;
+  let countBol = 0;
+  
+  if (wordsArr.length === 0) {
+    return null;
+  } else {
+    for (let i = 0; i < wordsArr.length; i++) {
+      let element = wordsArr[i];
+      if (typeof element === 'number') {
+        countNum += element;
+      } else if (typeof element === 'string') {
+        countChar += element.length;
+      } else if (typeof element === 'boolean'){
+        countBol += element;
+      } 
+    }
+  }
+  let sumNumChar = countChar + countNum + countBol;
+  return sumNumChar / wordsArr.length;
```

Comment:
```suggestion
  return sum(wordsArr)/ wordsArr.length;
```

Fijate!, has repetido todo el codigo que estaba en sum, pues entonces para ver la potencia de las funciones ejecutamos sum, repito, ejecutamos pasandole como argumento el array y listo!, tendriamos lo mismo!!! :D